napari/utils/colormaps/colormap_utils.py

```diff
@@ -29,20 +23,70 @@
 ]
 
 
-primary_color_names = ['red', 'green', 'blue', 'cyan', 'magenta', 'yellow']
-primary_colors = np.array(
-    [(1, 0, 0), (0, 1, 0), (0, 0, 1), (0, 1, 1), (1, 0, 1), (1, 1, 0)],
-    dtype=float,
+matplotlib_colormaps = _MATPLOTLIB_COLORMAP_NAMES = OrderedDict(
+    viridis=trans._('viridis'),
+    magma=trans._('magma'),
+    inferno=trans._('inferno'),
+    plasma=trans._('plasma'),
+    gray=trans._('gray'),
+    gray_r=trans._('gray r'),
+    hsv=trans._('hsv'),
+    turbo=trans._('turbo'),
+    twilight=trans._('twilight'),
+    twilight_shifted=trans._('twilight shifted'),
+    gist_earth=trans._('gist earth'),
+    PiYG=trans._('PiYG'),
+)
+_VISPY_CMAPS = get_colormaps()
+_VISPY_COLORMAPS = OrderedDict(
+    autumn=(trans._('autumn'), _VISPY_CMAPS['autumn']),
+    blues=(trans._('blues'), _VISPY_CMAPS['blues']),
+    cool=(trans._('cool'), _VISPY_CMAPS['cool']),
+    greens=(trans._('greens'), _VISPY_CMAPS['greens']),
+    reds=(trans._('reds'), _VISPY_CMAPS['reds']),
+    spring=(trans._('spring'), _VISPY_CMAPS['spring']),
+    summer=(trans._('summer'), _VISPY_CMAPS['summer']),
+    fire=(trans._('fire'), _VISPY_CMAPS['fire']),
+    grays=(trans._('grays'), _VISPY_CMAPS['grays']),
+    hot=(trans._('hot'), _VISPY_CMAPS['hot']),
+    ice=(trans._('ice'), _VISPY_CMAPS['ice']),
+    winter=(trans._('winter'), _VISPY_CMAPS['winter']),
+    light_blues=(trans._('light blues'), _VISPY_CMAPS['light_blues']),
+    orange=(trans._('orange'), _VISPY_CMAPS['orange']),
+    viridis=(trans._('viridis'), _VISPY_CMAPS['viridis']),
+    coolwarm=(trans._('coolwarm'), _VISPY_CMAPS['coolwarm']),
+    PuGr=(trans._('PuGr'), _VISPY_CMAPS['PuGr']),
+    GrBu=(trans._('GrBu'), _VISPY_CMAPS['GrBu']),
+    GrBu_d=(trans._('GrBu_d'), _VISPY_CMAPS['GrBu_d']),
+    RdBu=(trans._('RdBu'), _VISPY_CMAPS['RdBu']),
+    cubehelix=(trans._('cubehelix'), _VISPY_CMAPS['cubehelix']),
+    single_hue=(trans._('single hue'), _VISPY_CMAPS['single_hue']),
+    hsl=(trans._('hsl'), _VISPY_CMAPS['hsl']),
+    husl=(trans._('husl'), _VISPY_CMAPS['husl']),
+    diverging=(trans._('diverging'), _VISPY_CMAPS['diverging']),
+    RdYeBuCy=(trans._('RdYeBuCy'), _VISPY_CMAPS['RdYeBuCy']),
+)
```

Comment:
Wait, are we adding all these colormaps to our list? I don't think we expose them all currently?

Reply:
This will also add additional maintenance jobs when vispy list of colormaps changed.

Reply:
>Wait, are we adding all these colormaps to our list? I don't think we expose them all currently?

I did not add anything that was not already exposed I believe.

>This will also add additional maintenance jobs when vispy list of colormaps changed.

Sure, that is why software needs to be maintained :) , also added a test that checks the number of colormaps available and compares with Vispy

Reply:
> Sure, that is why software needs to be maintained :)

that's not, in itself, a great reason.   certainly it's reasonable to question the maintenance burden for any given new feature?

Reply:
I was being cheeky :). The main issues stemmed from having things like:

`light_blues` in the interface which is [not very readable](https://github.com/napari/product-heuristics-2020/issues/14)

Which should be transformed to `light blues` and is completely translatable (in Spanish at least), `Azules claros` (to put an example)

Having all strings with `trans` moves the decision of what makes sense to translate from the developer to the translator, and yes some might not make sense to translate on some or all languages, but it should not be up to us to make that call as it is perfectly fine to leave the string as the same on a given language (over at crowdin)

Now, having these colormaps from other modules and sources does imply that we need to maintain this now as well, hence why I try to add [a test that will fail as soon as an extra thing comes from Vispy](https://github.com/napari/napari/pull/2498/files#diff-1ad08d35dc40dc902c7f9417acee128bcb067edb46b133c48aba7cdf8b4e68abR132) that will reduce the burden, or ate least make things apparent.

Reply:
But when translating this you lost compatibility with the rest of the scientific software world. It may not be descriptive for persons who start their scientific journey from napari, but another one which starts from other tools then may have a big problem to find ones which they used in other places. 

Also, I could imagine a situation when a person with Spanish localization writes a script which deals with napari, and this script will not work for person from Deucheland because the first one uses Spanish name instead of English one... 

The case when person use different name in interface (Spanish) and code (English).

Reply:
>Also, I could imagine a situation when a person with Spanish localization writes a script which deals with napari, and this script will not work for person from Deucheland because the first one uses Spanish name instead of English one...

Colormaps have a `name` and a `display_name`, the name is and will continue to be the english original name in the example above `light_blues`, so this issue will not occur.

We could display the original `name` in parens, if that makes more sense in the tooltip or similar.

Reply:
thanks for that explanation @goanpeca - I've tried to expand on it here https://github.com/napari/napari/pull/2498#pullrequestreview-630207269, but it is very important to note this PR is backwards compatible and all our scripts from before should continue to work in all languages with `layer.colormap.name`

I am thinking now @goanpeca though, maybe we should make `colormap._display_name` private so that people don't modify it or look at directly themselves. I think that would be a good idea to assuage any of the concerns here. Thoughts?

Reply:
Fine by me!

Reply:
Actually... not so sur enow :-|

Reply:
@goanpeca 

> I did not add anything that was not already exposed I believe.

I think the question can be more accurately expressed as: we don't actually expose most of those colormaps in the UI (neither in master nor in this PR) so why are we making this giant list?

> Actually... not so sur enow :-|

... because? 😅

Reply:
Fixed it anyway 🤷🏼  :)